src/summarizedexperiment/utils.py

```diff
@@ -0,0 +1,80 @@
+from typing import Optional, Sequence, MutableMapping
+from functools import reduce
+import numpy as np
+import pandas as pd
+import scipy.sparse as sp
+
+from .SummarizedExperiment import SummarizedExperiment
+
+
+def concat(
+    ses: Sequence[SummarizedExperiment],
+    how: str = "outer",
+    metadata: Optional[MutableMapping] = None,
+) -> SummarizedExperiment:
+    """Concatenate SummarizedExperiment objects.
+    Assume we are working with RNA-Seq experiments for now.
+
+    Parameters
+    ----------
+    ses: Sequence[SummarizedExperiment]
+        A sequence of SummarizedExperiment objects.
+    how: str
+        Method to join SummarizedExperiment objects.
+    metadata: str
+        Metadata of the new SummarizedExperiment.
+
+    Returns
+    -------
+    concat_se: SummarizedExperiment
+        The concatenated SummarizedExperiment.
+    """
+
+    # we assume that sces have the same assay structure (same number of rows and cols)
+    new_assays = {}
+    assay_names = ses[0].assays.keys()
+    for assay_name in assay_names:
+        curr_assays = []
+        for sce in ses:
+            curr_assay = sce.assays[assay_name]
+            curr_assays.append(
+                pd.DataFrame(
+                    curr_assay, columns=sce.colData.index, index=sce.rowData.index
+                )
+                if isinstance(curr_assay, np.ndarray)
+                else pd.DataFrame.sparse.from_spmatrix(
+                    curr_assay, columns=sce.colData.index, index=sce.rowData.index
+                )
+            )
+
+        merged_assays = reduce(
+            lambda left, right: pd.merge(
+                left, right, left_index=True, right_index=True, how=how
+            ),
+            curr_assays,
+        )
+        merged_assays = merged_assays.sort_index().sort_values(
+            by=merged_assays.columns.tolist()
+        )
+        merged_assays = sp.csr_matrix(merged_assays.values)
+        new_assays[assay_name] = merged_assays
```

Comment:
this might become much simpler, once we identify what clarify the operations. (see comment on `rbind, cbind, ...`)

Reply:
what do you think of adding an `axis` argument which allows the caller to choose whether they want something more like combineRows or combineCols?